File: apps/web/app/office/lib/routine-status.test.ts
import { describe, expect, it } from "vitest";

import { isRoutineFiring } from "./routine-status";

describe("isRoutineFiring", () => {
  it("fires for active and the empty string", () => {
    expect(isRoutineFiring("active")).toBe(true);
    expect(isRoutineFiring("")).toBe(true);
  });

  it("does not fire for paused or archived", () => {
    expect(isRoutineFiring("paused")).toBe(false);
    expect(isRoutineFiring("archived")).toBe(false);
  });

  it("does not fire for an unrecognized value", () => {
    expect(isRoutineFiring("on_hold")).toBe(false);
  });

  it("is byte-exact: no case folding, no trimming", () => {
    expect(isRoutineFiring("Active")).toBe(false);
    expect(isRoutineFiring(" active")).toBe(false);
    expect(isRoutineFiring("active ")).toBe(false);
  });
});

File: apps/backend/internal/office/models/routine_status_test.go
package models

import "testing"

// TestRoutineStatusCanFire covers AC-OFFICE-ROUTINE-STATUS-001.1 through
// -001.4: an allowlist, not a denylist, and a byte-exact comparison with no
// case folding and no trimming.
func TestRoutineStatusCanFire(t *testing.T) {
	tests := []struct {
		name   string
		status RoutineStatus
		want   bool
	}{
		{"active fires", RoutineStatusActive, true},
		{"empty string fires", "", true},
		{"paused suppresses", RoutineStatusPaused, false},
		{"archived suppresses", RoutineStatusArchived, false},
		{"unrecognized value suppresses", "some_future_status", false},
		{"different case does not fire", "Active", false},
		{"leading space does not fire", " active", false},
		{"trailing space does not fire", "active ", false},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := tt.status.CanFire(); got != tt.want {
				t.Errorf("RoutineStatus(%q).CanFire() = %v, want %v", string(tt.status), got, tt.want)
			}
		})
	}
}
